tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/core/SentenceTransform.scala

```diff
@@ -161,9 +164,106 @@ case class VerbnetTagger(verbnet: Verbnet) extends SentenceTransform {
+      case (tagged, untagged) =>
+        val depLabelFreqMap = GoogleUnigram.getDepLabelNormalizedDistribution(
+          tagged, googleNgram.ngramMap, googleNgram.frequencyCutoff
+        )
+        // Create feature for each dependency label based on the normalized frequency
+        // bucket it lies in.
+        val frequencyFeatureMap: Map[Symbol, Set[Symbol]] = (for {
+          depLabel <- depLabelFreqMap.keySet
+        } yield {
+          val normalizedFrequency = depLabelFreqMap(depLabel)
+          val symbolSetWithCurrentDepLabel = Set(Symbol(depLabel))
+          if (normalizedFrequency <= 0.0009) {
+            ('depLabelFreqSmall, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.01) {
+            ('depLabelFreqBelow1, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.1) {
+            ('depLabelFreq1to10, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.2) {
+            ('depLabelFreq11to20, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.3) {
+            ('depLabelFreq21to30, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.4) {
+            ('depLabelFreq31to40, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.5) {
+            ('depLabelFreq41to50, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.6) {
+            ('depLabelFreq51to60, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.7) {
+            ('depLabelFreq61to70, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.8) {
+            ('depLabelFreq71to80, symbolSetWithCurrentDepLabel)
+          } else if (normalizedFrequency <= 0.9) {
+            ('depLabelFreq81to90, symbolSetWithCurrentDepLabel)
+          } else {
+            ('depLabelFreq91to100, symbolSetWithCurrentDepLabel)
+          }
+        }).groupBy(_._1).mapValues(_.flatMap(v => v._2))
+        untagged.updateProperties(frequencyFeatureMap)
+    }))
+  }
+}
+
+/** Frequency Distribution of POS tags for words based on Google Ngram's Nodes (unigrams).
+  */
+case class GoogleUnigramPostagTagger(googleNgram: GoogleNGram) extends SentenceTransform {
+
+  override def transform(sentence: Sentence): Sentence = {
+    Sentence(NexusToken +: ((sentence.tokens.tail) map {
+      tok =>
+        val postagFreqMap = GoogleUnigram.getPosTagNormalizedDistribution(
+          tok.word.name, googleNgram.ngramMap, googleNgram.frequencyCutoff
+        )
+        // Create feature for each dependency label based on the normalized frequency
+        // bucket it lies in.
+        val frequencyFeatureMap: Map[Symbol, Set[Symbol]] = (for {
+          postag <- postagFreqMap.keySet
+        } yield {
+          val normalizedFrequency = postagFreqMap(postag)
+          val symbolSetWithCurrentPostag = Set(Symbol(postag))
+          if (normalizedFrequency <= 0.0009) {
```

Comment:
Probably you don't want this many buckets. It likely just confuses the learner to have to treat 50-60% completely separately from 60-70%. I would suggest a bucketing more like the following: 1-5% 5-20% 20-50% 50-95% 95+%.